afl-analyze.c

```diff
@@ -28,6 +28,9 @@
 */
 
 #define AFL_MAIN
+#ifdef __ANDROID__
```

Comment:
Instead of doing
```
#ifdef __ANDROID__
  #include "android-ashmem.h"
#endif
```
where it might be needed, how about just including it everywhere and then putting the entire android-ashmem.h file in:
```
#ifdef __ANDROID__
  ...
#endif
```

Reply:
Can you explain more, I didn't get the point.

Reply:
I think it is ugly to do
```
#ifdef __ANDROID__
  #include "android-ashmem.h"
#endif 
```
in every file that mst include "android-ashmem.h". So instead just do 
```
#include "android-ashmem.h"
```
In these files.

And then in android-ashmem.h, put this around the entire file:
```
#ifdef __ANDROID__
  ...
#endif 
```
So that the header file is empty on non-android but can be included anywhere.
Here's an example of this from libFuzzer:
Ordinarily [this file](https://cs.chromium.org/chromium/src/third_party/libFuzzer/src/FuzzerExtFunctionsWindows.cpp?l=13) can't compile on non-Windows, but it's OK to build it because the entire file is wrapped in
```
#ifdef LIBFUZZER_WINDOWS
...
#endif 
```

Does this make more sense? If not I can try to explain again, or I can fix this myself after we land this code.